contrib/babelfishpg_common/sql/geometry.sql

```diff
@@ -142,17 +215,39 @@ CREATE OR REPLACE FUNCTION sys.Geometry__STPointFromText(text, integer)
 	AS $$
 	DECLARE
 		srid integer;
+		Geomtype text;
+		geom sys.GEOMETRY;
 	BEGIN
 		srid := $2;
 		IF srid >= 0 AND srid <= 999999 THEN
 			-- Call the underlying function after preprocessing
-			RETURN (SELECT sys.stgeomfromtext_helper($1, $2));
+			geom = (SELECT sys.stgeomfromtext_helper($1, $2));
+			Geomtype = (SELECT sys.ST_GeometryType(geom));
+			IF Geomtype = 'ST_Point' THEN
+				IF (SELECT sys.ST_Zmflag(geom)) = 1 OR (SELECT sys.ST_Zmflag(geom)) = 2 OR (SELECT sys.ST_Zmflag(geom)) = 3 THEN
+					RAISE EXCEPTION 'Z and M flags are currently not supported';
+				ELSE
+					RETURN geom;
+				END IF;
+			ELSE
+				RAISE EXCEPTION '% is currently not supported', Geomtype;
```

Comment:
'%s is not supported'. IMO we should remove the "currently" lines from the exceptions because we don't want to make promises about when things will be supported.

Reply:
Done